lib/sorbet-rails/gem_plugins/flag_shih_tzu_plugin.rb

```diff
@@ -0,0 +1,183 @@
+    return unless @model_class.include?(::FlagShihTzu)
+
+    obj_custom_module_name = self.model_module_name("GeneratedFlagShihTzuMethods")
+    obj_custom_module_rbi  = root.create_module(obj_custom_module_name)
+
+    klass_custom_module_name = self.model_module_name("GeneratedFlagShihTzuClassMethods")
+    klass_custom_module_rbi  = root.create_module(klass_custom_module_name)
+
+
+    # here we re-create the model class!
+    model_class_rbi = root.create_class(self.model_class_name)
+    model_class_rbi.create_include(obj_custom_module_name)
+    model_class_rbi.create_extend(klass_custom_module_name)
+
+    # then create custom methods, constants, etc. for this module.
+    add_class_methods(klass_custom_module_rbi)
+
+    if @model_class.flag_columns.present?
+      @model_class
+        .flag_columns
+        .each do |column|
+          add_methods_for_column(column, obj_custom_module_rbi)
+        end
+    end
+
+    if @model_class.flag_mapping.present?
+      @model_class
+        .flag_mapping
+        .each do |column, flags|
+          flags.keys
+            .select { |flag_key| really_has_the_flag?(flag_key) }
+            .each  do |flag_key|
+              add_methods_for_flag(column, flag_key, obj_custom_module_rbi)
+            end
+        end
+    end
+  end
+
+  private
+
+  sig { params(custom_module_rbi: Parlour::RbiGenerator::ModuleNamespace).void }
+  def add_class_methods(custom_module_rbi)
+    # https://github.com/pboling/flag_shih_tzu/blob/6a3f1c5f62bd56aa932252eef935826c9674b096/lib/flag_shih_tzu.rb#L12
+    custom_module_rbi.create_method('flag_options',
+                                    returns: 'T::Hash[String, T::Hash[Symbol, T.untyped]]')
+
+    # https://github.com/pboling/flag_shih_tzu/blob/6a3f1c5f62bd56aa932252eef935826c9674b096/lib/flag_shih_tzu.rb#L13
+    custom_module_rbi.create_method('flag_mapping',
+                                    returns: 'T::Hash[String, T::Hash[Symbol, Integer]]')
+
+    # https://github.com/pboling/flag_shih_tzu/blob/6a3f1c5f62bd56aa932252eef935826c9674b096/lib/flag_shih_tzu.rb#L14
+    custom_module_rbi.create_method('flag_columns',
+                                    returns: 'T::Array[Symbol]')
+
+    # https://github.com/pboling/flag_shih_tzu#updating-flag-column-by-raw-sql
+    custom_module_rbi.create_method('set_flag_sql',
+                                    parameters: [
+                                      ::Parlour::RbiGenerator::Parameter.new(
+                                        'flag',
+                                        type: 'Symbol'
+                                      ),
+                                      ::Parlour::RbiGenerator::Parameter.new(
+                                        'value',
+                                        type: 'T::Boolean'
+                                      ),
+                                      ::Parlour::RbiGenerator::Parameter.new(
+                                        'colmn',
```

Comment:
typo: `column`

Reply:
It's funny because is not a typo. The [method definition](https://github.com/pboling/flag_shih_tzu/blob/6a3f1c5f62bd56aa932252eef935826c9674b096/lib/flag_shih_tzu.rb#L270):

```ruby
def set_flag_sql(flag, value, colmn = nil, custom_table_name = table_name)
```

Reply:
Ouch lol. Thanks for clarifying!